test/framework/src/main/java/org/elasticsearch/common/util/concurrent/DeterministicTaskQueue.java

```diff
@@ -380,7 +317,12 @@ public ExecutorService generic() {
 
             @Override
             public ExecutorService executor(String name) {
-                return forkingExecutor;
+                final CachedSupplier<ExecutorService> supplier = predefinedExecutors.get(name);
+                if (supplier != null) {
+                    return supplier.get();
```

Comment:
This might be controvesial. It allows a set of executors to run tasks without going through `DeterministicTaskQueue`. They are used to allow blocking reads to work.

Reply:
We discussed offline and David has a great suggestion of using the direct executor for such blocking calls so that the future is ready when the blocking happens. I have updated the implementation based on this idea.